cpp/src/join/join.cu

```diff
@@ -331,11 +331,11 @@ construct_join_output_df(
 
   std::unique_ptr<experimental::table> common_table;
```

Comment:
I feel like it would be cleaner to initialize `common_table `to `std::make_unique<experimental::table>()` and eliminate the added else branches. As it is, the code/logic is repeated in the two branches below.

Reply:
will do